### PR TITLE
StatisticsAggregator returns original tensor shape

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Make StatisticsAggreagtor keep the original tensor share after aggregation.

### Reason for changes

To add support of correct handling statistics in case batch_size > 1.

### Related tickets

121650

### Tests

All tests are updated accordingly

### CODE REVIEW

File: nncf/experimental/common/tensor_statistics/collectors.py [hunk 1]
```diff
@@ -734,32 +728,30 @@ def _aggregation_fn(
...
-
         mad_values = self._tensor_processor.median(
             self._tensor_processor.abs(self._tensor_processor.sub(stacked_val, median_per_ch)),
             axis=self._aggregation_axes,
             keepdims=self._keepdims,
         )
-        if not self._keepdims:
-            median_per_ch = self._tensor_processor.squeeze(median_per_ch, self._aggregation_axes)
+        squeezed_mad_values = self._tensor_processor.squeeze(mad_values, 0)
+        squeezed_median_per_ch = self._tensor_processor.squeeze(median_per_ch, 0)
```
Comment (daniil-lyakhov) [inline nncf/experimental/common/tensor_statistics/collectors.py:747]: Please leave branch with `not self._keepdims`, I'm planning to expand testing with different `self._keepdims` values, so next time we will need to change it we shouldn't change any test

Comment (kshpv) [reply nncf/experimental/common/tensor_statistics/collectors.py]: Rebased, done

**Review by daniil-lyakhov — APPROVED** [review]

File: nncf/experimental/common/tensor_statistics/collectors.py [hunk 2]
```diff
@@ -622,19 +611,17 @@ def _register_reduced_input_impl(self, x: NNCFTensor) -> None:
...
         if 0 in self._aggregation_axes:
-            if self._container:
-                reduced = self._aggregation_fn(
-                    self._tensor_processor.stack([reduced, self._container]), axis=0, keepdims=False
-                )
-            self._container = reduced
+            stacked_tensors = self._tensor_processor.stack([reduced, *self._container], axis=0)
+            aggregated = self._aggregation_fn(stacked_tensors, axis=0, keepdims=self._keepdims)
+            aggregated = self._tensor_processor.squeeze(aggregated, 0)
+            self._container = [aggregated]
```
Comment (KodiaqQ) [inline nncf/experimental/common/tensor_statistics/collectors.py:617]: Could you please provide the example for the container size/shape transformation?

File: tests/common/experimental/test_reducers_and_aggregators.py [hunk 3]
```diff
@@ -254,38 +252,38 @@ def test_min_max_aggregators(
         assert self.all_close(max_aggregator.aggregate(), max_ref)
 
     NO_OUTLIERS_TEST_PARAMS = [
-        (MeanAggregator, True, 1, 1404.5138888888905),
-        (MedianAggregator, True, 1, 24.0),
+        (MeanAggregator, True, 1, [1404.5138888888905]),
+        (MedianAggregator, True, 1, [24.0]),
```
Comment (KodiaqQ) [inline tests/common/experimental/test_reducers_and_aggregators.py:256]: Is it possible with the changes that scalar statistics stay as scalar?

Comment (kshpv) [reply tests/common/experimental/test_reducers_and_aggregators.py:256]: The logic should be the same for all tensors: to keep dims of the original tensor.
In that test input tensor has `dim=1`

File: nncf/experimental/common/tensor_statistics/collectors.py [hunk 4]
```diff
@@ -622,19 +611,17 @@ def _register_reduced_input_impl(self, x: NNCFTensor) -> None:
...
         else:
             self._container.append(reduced)
 
     def _aggregate_impl(self) -> NNCFTensor:
         if 0 in self._aggregation_axes:
             if self._keepdims:
-                return self._tensor_processor.stack([self._container]).tensor
-            return self._container.tensor
+                return self._container[0].tensor
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/experimental/common/tensor_statistics/collectors.py:624]: I wanted to have the same datatype for all cases. Before depending on parameters it was `List` or `Tensor`.

File: nncf/experimental/common/tensor_statistics/collectors.py [hunk 5]
```diff
@@ -125,6 +125,7 @@ def __init__(
         tensor_processor: NNCFCollectorTensorProcessor,
         aggregation_axes: Optional[AggregationAxes] = None,
         num_samples: Optional[int] = None,
+        window_size=None,
```
[the remark this answers is not in the document]
Comment (kshpv) [reply nncf/experimental/common/tensor_statistics/collectors.py:128]: Updated

Comment (KodiaqQ) [inline nncf/experimental/common/tensor_statistics/collectors.py:128]:
```suggestion
        window_size: Optional[..] = None,
```

**Review by KodiaqQ — APPROVED** [review]

File: nncf/experimental/common/tensor_statistics/collectors.py (hunk 5 above)
Comment (kshpv) [reply nncf/experimental/common/tensor_statistics/collectors.py]: Applied

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**